shell/platform/darwin/macos/framework/Source/FlutterMenuPlugin.mm

```diff
@@ -0,0 +1,418 @@
+
+    // As copied, these platform provided menu items don't yet have the APP_NAME
+    // string replaced in them, so this rectifies that.
+    [self replaceAppName:_platformProvidedItems];
+  }
+  return self;
+}
+
+/**
+ * Iterates through the given menu hierarchy, and replaces "APP_NAME"
+ * with the localized running application name.
+ */
+- (void)replaceAppName:(NSArray<NSMenuItem*>*)items {
+  NSString* appName = [NSRunningApplication currentApplication].localizedName;
+  for (NSMenuItem* item in items) {
+    if ([[item title] containsString:kAppName]) {
+      [item setTitle:[[item title] stringByReplacingOccurrencesOfString:kAppName
+                                                             withString:appName]];
+    }
+    if ([item hasSubmenu]) {
+      [self replaceAppName:[[item submenu] itemArray]];
+    }
+  }
+}
+
+- (NSMenuItem*)findProvidedMenu:(NSMenu*)menu ofType:(SEL)selector withTag:(int)tag {
+  const NSArray<NSMenuItem*>* items = menu != nil ? menu.itemArray : _platformProvidedItems;
+  for (NSMenuItem* item in items) {
+    if ([item action] == selector && (tag == 0 || [item tag] == tag)) {
+      return item;
+    }
+    if ([[item submenu] numberOfItems] > 0) {
+      NSMenuItem* foundChild = [self findProvidedMenu:[item submenu] ofType:selector withTag:tag];
+      if (foundChild != nil) {
+        return foundChild;
+      }
+    }
+  }
+  return nil;
+}
+
+- (NSMenuItem*)createPlatformProvidedMenu:(PlatformProvidedMenu)type {
+  const std::map<int, SEL> providedMenus = GetMacOSProvidedMenus();
+  auto found_type = providedMenus.find(type);
+  if (found_type == providedMenus.end()) {
+    return nil;
+  }
+  SEL selectorTarget = found_type->second;
+  // Since it doesn't have a definitive selector, the Services submenu is
+  // assumed to be the first item with a submenu action in the first menu item
+  // of the default menu set. We can't just get the title to check, since that
+  // is localized, and the contents of the menu aren't fixed (or even available).
+  NSMenu* startingMenu =
+      type == PlatformProvidedMenu::servicesSubmenu ? [_platformProvidedItems[0] submenu] : nil;
+  NSMenuItem* found = [self findProvidedMenu:startingMenu ofType:selectorTarget withTag:0];
+  if (found != nil) {
+    // Return a copy because the original menu item might not have been removed
+    // from the main menu yet, and AppKit doesn't like menu items that exist in
+    // more than one menu at a time.
+    return [found copy];
+  }
+  return nil;
+}
+
+- (NSMenuItem*)menuItemFromFlutterRepresentation:(NSDictionary*)representation {
+  if ([(NSNumber*)([representation valueForKey:kDividerKey]) intValue] == YES) {
+    return [NSMenuItem separatorItem];
+  }
+  NSString* appName = [NSRunningApplication currentApplication].localizedName;
+  NSString* title = representation[kLabelKey];
+  if (title != nil) {
+    title = [title stringByReplacingOccurrencesOfString:kAppName withString:appName];
+  }
```

Comment:
Similarly here you can skip the nil check. This can be:
```objc
title = [title stringByReplacingOccurrencesOfString:kAppName withString:appName];
```
or you can tack the message onto the end of the initialisation above.

Reply:
Done.